### ISSUE TITLE
Deriving strategies

### ISSUE THREAD

**Issue by adamgundry** [iconnect/api-tools#68]
At the moment, `api-tools` uses derived instances for its generated types, but does not specify the deriving strategy. This can lead to problems, in particular for newtypes when both `DeriveAnyClass` and `GeneralizedNewtypeDeriving` are enabled, because the choice of strategy is ambiguous. Ideally we would explicitly request `newtypeStrategy` to avoid this; although that might require the `DerivingStrategies` extension to be enabled in user code.

**Comment by cdornan** [comment]
I agree &mdash; best to be explicit, even at the cost of requiring `DerivingStrategies`.

**Comment by adamgundry** [comment]
FWIW it looks like TH code can test which extensions are enabled, so it would be possible to do this in a backwards-compatible way: require `DerivingStrategies` only if `DeriveAnyClass` is enabled.
